Tests/SmithyRetriesTests/RetryIntegrationTests.swift

```diff
@@ -54,7 +54,13 @@ final class RetryIntegrationTests: XCTestCase {
             .attributes(context)
             .retryErrorInfoProvider(DefaultRetryErrorInfoProvider.errorInfo(for:))
             .retryStrategy(subject)
-            .deserialize({ _, _ in TestOutputResponse() })
+            .deserialize({ response, _ in
+                if response.statusCode == .ok {
+                    return TestOutputResponse()
+                } else {
+                    throw TestHTTPError(statusCode: response.statusCode)
+                }
+            })
```

Comment:
The deserializer above and the switch statement below were corrected to ensure that a HTTP response is available during retries.